src/Elastic.Apm/OpenTelemetry/ElasticSearchHttpNonTracer.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+#if NET5_0_OR_GREATER
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Elastic.Apm.Api;
+using Elastic.Apm.DiagnosticListeners;
+
+namespace Elastic.Apm.OpenTelemetry
+{
+	/// <summary>
+	/// Handles HTTP spans for outgoing HTTP calls from `Elastic.Clients.ElasticSearch`.
+	/// Since `Elastic.Clients.ElasticSearch` emits <see cref="Activity"/>, and according to our spec, for calls into elasticsearch
+	/// we don't need to create an extra HTTP span, all this does is that it suppresses span creation during the outgoing HTTP call.
+	/// </summary>
+	public class ElasticSearchHttpNonTracer : IHttpSpanTracer
+	{
+		public bool IsMatch(string method, Uri requestUrl, Func<string, string> headerGetter) => false;
+		public ISpan StartSpan(IApmAgent agent, string method, Uri requestUrl, Func<string, string> headerGetter) => null;
+
+		public bool SuppressSpanCreation()
+		{
+			if (Activity.Current == null || Activity.Current.Parent == null)
+				return false;
+			return Activity.Current.Parent.DisplayName.StartsWith("Elasticsearch:") && Activity.Current.Parent.Tags.Any(n => n is { Key: "db.system", Value: "elasticsearch" });
```

Comment:
There are cases where `SuppressSpanCreation` returns `false`,  however `IsMatch` always returns `false` and `StartSpan` is a no-op. Is this intentional?

Reply:
Yes, that is intentional and (from my understanding) is the expected behaviour we want.

`SuppressSpanCreation` returns `false` means we deal with an http call and it's not coming from the Elasticsearch (es) client (e.g. a normal call via `HttpClient` to some web service).

`IsMatch` always returns `false`: so if it's not from the es client, the `ElasticSearchHttpNonTracer.cs` never matches any any outgoing HTTP call. In `HttpDiagnosticListenerImplBase` you can see, if `IsMatch` returns `false`, then the normal outgoing HTTP capturing will run (so we capture the call as a normal http span).

`StartSpan` is a no-op: this will never run because `IsMatch` returns false. And even if it runs, we return `null` which makes the normal HTTP capturing to kick in.

So in case of an HTTP call, the result is:
- if it's from the es client: in `HttpDiagnosticListenerImplBase` we suppress span creation completely - no span for those http calls.
- if it's not not from the es client: it'll never match any url, so `HttpDiagnosticListenerImplBase` will run the normal HTTP span creation code path.

Reply:
Makes sense, thanks for the explanation 👍.